qcodes/utils/dataset/doNd.py

```diff
@@ -1,1054 +1,23 @@
-            all_meas_parameters.append(param)
-            if isinstance(param, _BaseParameter):
-                measured_parameters.append(param)
-        elif not isinstance(param, str):
-            multi_group.append(param)
-            for nested_param in param:
-                all_meas_parameters.append(nested_param)
-                if isinstance(nested_param, _BaseParameter):
-                    measured_parameters.append(nested_param)
-    if single_group:
-        pg: ParameterGroup = {
-            "params": tuple(single_group),
-            "meas_name": measurement_name,
-            "measured_params": [],
-        }
-        grouped_parameters["group_0"] = pg
-    if multi_group:
-        for index, par in enumerate(multi_group):
-            pg = {
-                "params": tuple(par),
-                "meas_name": measurement_name,
-                "measured_params": [],
-            }
-            grouped_parameters[f"group_{index}"] = pg
-    return tuple(all_meas_parameters), grouped_parameters, tuple(measured_parameters)
-
-
-def _handle_plotting(
-    data: DataSetProtocol,
-    do_plot: bool = True,
-    interrupted: MeasInterruptT = None,
-) -> AxesTupleListWithDataSet:
-    """
-    Save the plots created by datasaver as pdf and png
-
-    Args:
-        datasaver: a measurement datasaver that contains a dataset to be saved
-            as plot.
-            :param do_plot:
-
-    """
-    if do_plot:
-        res = plot(data)
-    else:
-        res = data, [None], [None]
-
-    if interrupted:
-        raise interrupted
-
-    return res
-
-
-def plot(
-    data: DataSetProtocol, save_pdf: bool = True, save_png: bool = True
-) -> Tuple[
-    DataSetProtocol,
-    List[matplotlib.axes.Axes],
-    List[Optional[matplotlib.colorbar.Colorbar]],
-]:
-    """
-    The utility function to plot results and save the figures either in pdf or
-    png or both formats.
-
-    Args:
-        data: The QCoDeS dataset to be plotted.
-        save_pdf: Save figure in pdf format.
-        save_png: Save figure in png format.
-    """
-    dataid = data.captured_run_id
-    axes, cbs = plot_dataset(data)
-    mainfolder = config.user.mainfolder
-    experiment_name = data.exp_name
-    sample_name = data.sample_name
-    storage_dir = os.path.join(mainfolder, experiment_name, sample_name)
-    os.makedirs(storage_dir, exist_ok=True)
-    png_dir = os.path.join(storage_dir, "png")
-    pdf_dif = os.path.join(storage_dir, "pdf")
-    os.makedirs(png_dir, exist_ok=True)
-    os.makedirs(pdf_dif, exist_ok=True)
-    for i, ax in enumerate(axes):
-        if save_pdf:
-            full_path = os.path.join(pdf_dif, f"{dataid}_{i}.pdf")
-            ax.figure.savefig(full_path, dpi=500)
-        if save_png:
-            full_path = os.path.join(png_dir, f"{dataid}_{i}.png")
-            ax.figure.savefig(full_path, dpi=500)
-    res = data, axes, cbs
-    return res
+# from qcodes.dataset.doNd import (
```

Comment:
i agree we should keep these imports working, but we can add a deprecation warning that triggers whenever an import from here is made.

Reply:
Will be added after this is released. See comment above